tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -370,20 +378,26 @@ def start_all_nodes_with(self, version):
                         third_monitor.wait_until(self.processed_data_msg,
                                                  timeout_sec=60,
                                                  err_msg="Never saw output '%s' on " % self.processed_data_msg + str(node3.account))
-                        if KafkaVersion(version).supports_fk_joins():
+                        if 'test.run_fk_join' in extra_properties:
```

Comment:
nit: let's also check that the corresponding value is `True`

```suggestion
                        if extra_properties.get('test.run_fk_join', False):
```